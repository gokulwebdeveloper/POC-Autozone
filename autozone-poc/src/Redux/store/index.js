import {createStore, applyMiddleware, compose} from 'redux';
import rootReducer from '../reducers';
import rootSaga from '../middlewares/saga';
import createSagaMiddleware from 'redux-saga';

// create the saga middleware
export const sagaMiddleware = createSagaMiddleware();

//list of middlewares
const middlewares = [sagaMiddleware];

//apply middleware on store
const store = createStore(rootReducer, applyMiddleware(...middlewares));

//run saga generator function
sagaMiddleware.run(rootSaga);

export default store;