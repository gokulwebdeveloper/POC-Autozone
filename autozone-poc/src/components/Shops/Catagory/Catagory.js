import React, { Fragment } from "react";
import ListItem from "../ListItem/ListItem";
import PropTypes from 'prop-types';


Catagory.propTypes = {
    data: PropTypes.string,
    data: PropTypes.any.isRequired,
}

export default function Catagory(props) {
    if (props != undefined && props.data != undefined) {
        return (
            <Fragment>
                <li data-toggle="collapse" data-target={props.datatarget}>
                    <a href="#">{props.data.title}</a>
                    <ul className="sub-menu collapse hide" id={props.id}>
                        <ListItem items={props.data.items} />
                    </ul>
                </li>
            </Fragment>
        );
    } else {
        return <p>-</p>
    }

}