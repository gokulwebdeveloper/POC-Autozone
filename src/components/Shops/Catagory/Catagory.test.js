import '@testing-library/jest-dom/extend-expect';
import renderer from "react-test-renderer";
import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Catagory from './Catagory';
import store from '../../../Redux/store';
import { Provider } from 'react-redux';

configure({ adapter: new Adapter() });


describe('Catagory testing', () => {
	var categoryList = [{
		"title": "clothing",
		"items": [
			"All",
			"Bodysuits",
			"Dresses",
			"Hoodies Sweats",
			"Jackets Coats",
			"Jeans",
			"Pants Leggings",
			"Rompers Jumpsuits",
			"Shirts Blouses",
			"Shirts",
			"Sweaters &amp; Knits"
		]
	}
	];


	let wrapper;
	wrapper = mount(<Provider store={store}><Catagory data={categoryList[0]}  id="clothing" /></Provider>);

	it("SnapShot Match for Catagory component", () => {
		const page = renderer.create(<Provider store={store}><Catagory data={categoryList[0]}
			 id="clothing" /></Provider>).toJSON();
		expect(page).toMatchSnapshot();
	});

	test('renders ', () => {
		expect(wrapper.find('li').at(2).text()).toBe('Bodysuits');
	});

	test('renders ListItem ', () => {
		expect(wrapper.find("ListItem").prop('title')).toBe('clothing');
		expect(wrapper.find("ListItem").prop('items')).toBe(categoryList[0].items);
	});

})