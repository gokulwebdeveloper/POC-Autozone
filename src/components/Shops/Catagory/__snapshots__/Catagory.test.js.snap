// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catagory testing SnapShot Match for Catagory component 1`] = `
<li>
  <p
    data-target="#clothing"
    data-toggle="collapse"
  >
    clothing
  </p>
  <ul
    className="sub-menu show"
    id="clothing"
  >
    <li
      data-id="All"
      data-title="clothing"
      onClick={[Function]}
    >
      All
    </li>
    <li
      data-id="Bodysuits"
      data-title="clothing"
      onClick={[Function]}
    >
      Bodysuits
    </li>
    <li
      data-id="Dresses"
      data-title="clothing"
      onClick={[Function]}
    >
      Dresses
    </li>
    <li
      data-id="Hoodies Sweats"
      data-title="clothing"
      onClick={[Function]}
    >
      Hoodies Sweats
    </li>
    <li
      data-id="Jackets Coats"
      data-title="clothing"
      onClick={[Function]}
    >
      Jackets Coats
    </li>
    <li
      data-id="Jeans"
      data-title="clothing"
      onClick={[Function]}
    >
      Jeans
    </li>
    <li
      data-id="Pants Leggings"
      data-title="clothing"
      onClick={[Function]}
    >
      Pants Leggings
    </li>
    <li
      data-id="Rompers Jumpsuits"
      data-title="clothing"
      onClick={[Function]}
    >
      Rompers Jumpsuits
    </li>
    <li
      data-id="Shirts Blouses"
      data-title="clothing"
      onClick={[Function]}
    >
      Shirts Blouses
    </li>
    <li
      data-id="Shirts"
      data-title="clothing"
      onClick={[Function]}
    >
      Shirts
    </li>
    <li
      data-id="Sweaters &amp; Knits"
      data-title="clothing"
      onClick={[Function]}
    >
      Sweaters &amp; Knits
    </li>
  </ul>
</li>
`;
